docs/buildstrategies.md

```diff
@@ -410,7 +410,27 @@ Contrary to the strategy `spec.parameters`, you can use system parameters and th
-| `$(params.shp-output-image)`      | The URL of the image that the user wants to push as specified in the Build's `spec.output.image`, or the override from the BuildRun's `spec.output.image`. |
+| `$(params.shp-output-directory)` | The absolute path to a directory that the build strategy should store the image in. You can store a single tarball containing a single image, or an OCI image layout. |
+| `$(params.shp-output-image)`     | The URL of the image that the user wants to push as specified in the Build's `spec.output.image`, or the override from the BuildRun's `spec.output.image`. |
+| `$(parama.shp-output-insecure)`  | Flag that indicates if the output image's registry location is insecure, which means that it uses a certificated not signed by a certificate authority, or uses HTTP |
+
+### Output directory vs output image
+
+As a build strategy author, you decide whether you just build the image, or also push it to the container registry. You implicitly indicate how your strategy behaves by using, or not using the `$(params.shp-output-image)` system parameter:
+
+- If you DO NOT use `$(params.shp-output-image)`, then Shipwright assumes that your build strategy PUSHES the image. We call this a strategy-managed push.
+- If you DO use `$(params.shp-output-image)`, then Shipwright assumes that your build strategy does NOT PUSH the image. We call this a shipwright-managed push.
+
+When you use the `$(params.shp-output-image)` parameter, then Shipwright will also take care to set the [image-related system results](#system-results).
+
+If you are uncertain about how to implement your build strategy, then follow this guidance:
+
+1. If your build strategy tool is unable to locally store an image but always pushes it, then you must do the push operation. An example is the [Buildpacks strategy](../samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3_cr.yaml). You SHOULD respect the `$(params.shp-output-insecure)` parameter.
+2. If your build strategy tool can locally store an image, then the choice depends on how you expect your build users to make use of your strategy, and the nature of your strategy.
+   1. Some build strategies do not produce all layers of an image, but only a layer or some layers with the application, and use a common base image. An example is `ko`. Such base image layers are often already present in the destination registry (like in rebuilds). If the strategy can perform the push operation, then it can optimize the process and can omit the download of the base image when it is not required to push it. In case of a shipwright-manager push, the complete image must be locally store in `$(params.shp-output-directory)` which implies that a base image must always be downloaded.
```

Comment:
```suggestion
   1. Some build strategies do not produce all layers of an image, but only a layer or some layers within the application, and use a common base image. An example is `ko`. Such base image layers are often already present in the destination registry (like in rebuilds). If the strategy can perform the push operation, then it can optimize the process and can omit the download of the base image when it is not required to push it. In the case of a shipwright-manager push, the complete image must be locally stored in `$(params.shp-output-directory)`, which implies that a base image must always be downloaded.
```

Reply:
"within" is inaccurate, but I get your point. I rephrased it.